src/sentry/issues/grouptype.py

```diff
@@ -178,3 +200,61 @@ class ProfileJSONDecodeType(GroupType):
         ProfileFileIOGroupType.type_id,
     ]
 )
+
+
+def reduce_noise(
+    new_grouphashes: Set[str],
+    performance_problems_by_hash: Dict[str, PerformanceProblem],
+    project: Project,
+) -> Set[str]:
+
+    groups_to_ignore = set()
+    cluster_key = settings.SENTRY_ISSUE_PLATFORM_RATE_LIMITER_OPTIONS.get("cluster", "default")
+    client = redis.redis_clusters.get(cluster_key)
+
+    for new_grouphash in new_grouphashes:
+        group_type = performance_problems_by_hash[new_grouphash].type
+        noise_config = group_type.noise_config
+        if not noise_config:
+            continue
+
+        ignore_limit, expiry_time = noise_config.ignore_limit, noise_config.expiry_time
+
+        if ignore_limit and not should_create_group(
+            client, new_grouphash, group_type, ignore_limit, expiry_time, project
+        ):
+            groups_to_ignore.add(new_grouphash)
+
+    new_grouphashes = new_grouphashes - groups_to_ignore
+    return new_grouphashes
+
+
+@metrics.wraps("noise_reduction.should_create_group", sample_rate=1.0)
+def should_create_group(
```

Comment:
To some extent I wonder if this makes sense on the `GroupType`? Then usage could just be `grouptype.should_create_group(client, grouphash, project)`.

Potentially we don't even need to pass in client.

Reply:
I had that in a previous attempt but then was running into issues with the frozenset or something. Will see what options are working now.

Reply:
Ah yeah so if this is only going over one hash at a time I'm cool to pass in the client. Otherwise would rather instantiate it once

Reply:
OK so the issue is that you still have to pass the `GroupType` to the function because a `dataclass` is never actually initialized and it's basically a pure function and not a class method